src/bin/crontab.cpp

```diff
@@ -87,13 +87,52 @@ static auto run_generator(const char * op, const char * file_or_line, bool file_
 	}
 }
 
+static const bool want_colour = !*(std::getenv("NO_COLOR") ?: "") && (std::getenv("TERM") ?: ""sv).find("color"sv) != std::string_view::npos && isatty(1);
+static void blue(const char * line) {
+	if(want_colour)
+		std::fputs("\033[1;34m", stdout);
+	std::fputs(line, stdout);
+	if(want_colour)
+		std::fputs("\033[0m", stdout);
+}
+
 static auto translate(const char * line) -> int {
-	return run_generator("--translate", line, false, false);
+	close(3);
+	auto timer = vore::file::FILE<false>::tmpfile();
+	if(!timer)
+		return std::fprintf(stderr, "%s: %s\n", self, std::strerror(errno)), 1;
+	assert(fileno(timer) == 3);
+
+	close(4);
+	auto service = vore::file::FILE<false>::tmpfile();
+	if(!service)
+		return std::fprintf(stderr, "%s: %s\n", self, std::strerror(errno)), 1;
+	assert(fileno(service) == 4);
+
+	auto err = run_generator("--translate", line, false);
+	if(err)
+		return err;
+
+	blue("# /etc/systemd/system/$unit.timer\n");
+	std::rewind(timer);
+	copy_FILE(timer, stdout);
+	std::fputc('\n', stdout);
+
+	blue("# /etc/systemd/system/$unit.service\n");
+	std::rewind(service);
+	copy_FILE(service, stdout);
+	std::fflush(stdout);
+
+	std::rewind(timer);
+	std::rewind(service);
+	execlp("systemd-analyze", "systemd-analyze", "verify", "/dev/fd/3:input.timer", "/dev/fd/4:input.service", static_cast<const char *>(nullptr));
+	// ignore error, optional analysis
```

Comment:
Wow, this _does_ actually work without the `SYSTEMD_UNIT_PATH=` ?
(disclaimer: 1 maintain 100 packages for a custom proprietary distro overlay on top of Debian and I didn't got `systemd-analyze verify` at first working in our lintian-esque CI so I dropped the ball back then)

Reply:
I decided to try it out since I saw it in the manual – <https://manpages.debian.org/bookworm/systemd/systemd-analyze.1.en.html#systemd-analyze_verify_FILE...>, example 19 – and it works fine. I glimpsed at the strace yesterday and it appears to just copy the units to a tmpdir, then define `SYSTEMD_UNIT_PATH=` internally (at least when driven with colons).